src/runtime_src/xdp/profile/plugin/aie_trace/edge/aie_trace.cpp

```diff
@@ -282,90 +315,204 @@ namespace xdp {
-                                           const uint8_t channel1) 
+  bool AieTrace_EdgeImpl::isStreamSwitchPortEvent(const XAie_Events event)
+  {
+    // AIE tiles
+    if ((event > XAIE_EVENT_GROUP_STREAM_SWITCH_CORE) 
+        && (event < XAIE_EVENT_GROUP_BROADCAST_CORE))
+      return true;
+    // Interface tiles
+    if ((event > XAIE_EVENT_GROUP_STREAM_SWITCH_PL) 
+        && (event < XAIE_EVENT_GROUP_BROADCAST_A_PL))
+      return true;
+    // Memory tiles
+    if ((event > XAIE_EVENT_GROUP_STREAM_SWITCH_MEM_TILE) 
+        && (event < XAIE_EVENT_GROUP_MEMORY_CONFLICT_MEM_TILE))
+      return true;
+
+    return false;
+  }
+
+  // Configure stream switch ports for monitoring purposes
+  // NOTE: Used to monitor streams: trace, interfaces, and memory tiles
+  void AieTrace_EdgeImpl::configStreamSwitchPorts(XAie_DevInst* aieDevInst, const tile_type& tile,
+                                                  xaiefal::XAieTile& xaieTile, const XAie_LocType loc,
+                                                  const module_type type, const XAie_Events event,
+                                                  const int countnum, const std::string metricSet,
+                                                  const uint8_t channel)
+  {
+    // Only configure as needed: must be applicable event and only need at most two
+    if (!isStreamSwitchPortEvent(event) || (countnum > 1))
+      return;
+
+    auto switchPortRsc = xaieTile.sswitchPort();
+    auto ret = switchPortRsc->reserve();
+    if (ret != AieRC::XAIE_OK)
+      return;
+
+    uint32_t rscId = 0;
+    XAie_LocType tmpLoc;
+    XAie_ModuleType tmpMod;
+    switchPortRsc->getRscId(tmpLoc, tmpMod, rscId);
+
+    // AIE Tiles (e.g., trace streams)
+    if (type == module_type::core) {
+      // Define stream switch port to monitor core or memory trace
+      uint8_t traceSelect = (event == XAIE_EVENT_PORT_RUNNING_0_CORE) ? 0 : 1;
+      XAie_EventSelectStrmPort(aieDevInst, loc, rscId, XAIE_STRMSW_SLAVE, TRACE, traceSelect);
+      return;
+    }
+
+    // Interface tiles (e.g., PLIO, GMIO)
+    if (type == module_type::shim) {
+      // Grab slave/master and stream ID
+      // NOTE: stored in getTilesForProfiling() above
+      auto slaveOrMaster = (tile.itr_mem_col == 0) ? XAIE_STRMSW_SLAVE : XAIE_STRMSW_MASTER;
+      auto streamPortId  = static_cast<uint8_t>(tile.itr_mem_row);
+
+      // Define stream switch port to monitor interface 
+      XAie_EventSelectStrmPort(aieDevInst, loc, rscId, slaveOrMaster, SOUTH, streamPortId);
+      return;
+    }
+
+    // MEM tiles
+    if (metricSet.find("trace") != std::string::npos) {
+      XAie_EventSelectStrmPort(aieDevInst, loc, rscId, XAIE_STRMSW_SLAVE, TRACE, 0);
+    }
+    else {
+      auto slaveOrMaster = (metricSet.find("output") != std::string::npos) ?
+        XAIE_STRMSW_SLAVE : XAIE_STRMSW_MASTER;
+      XAie_EventSelectStrmPort(aieDevInst, loc, rscId, slaveOrMaster, DMA, channel);
```

Comment:
This part of the code totally doesn't make sense to me. We reserve a stream switch port and that gives us a resource ID and we map it to the appropriate channel. Why do we completely ignore this ID and trace some hardcoded IDs instead? This seems to completely circumvent the resource manager.

Reply:
@hackwa Thanks for catching this and the feedback. This will be addressed in a future PR.